### PR TITLE
fix(rust): honor Buildkite/CircleCI/Jenkins env in ci scopes-send

### PR DESCRIPTION
**Pull request by sileht**
The Python ``ci scopes-send`` resolves the repository and PR number
through ``detector.get_github_repository`` /
``get_github_pull_request_number``, which match
``get_ci_provider()`` and consult provider-specific env vars
(``BUILDKITE_REPO`` / ``BUILDKITE_PULL_REQUEST``,
``CIRCLE_REPOSITORY_URL``, ``GIT_URL``). The Rust port only read
``GITHUB_REPOSITORY`` and ``GITHUB_EVENT_PATH``, so on Buildkite it
errored on missing ``--repository`` (no ``GITHUB_REPOSITORY``) or
silently skipped because no PR number was detected — making
``mergify ci scopes-send`` unusable outside GitHub Actions.

This brings the Rust resolver in line:

- ``detect_ci_provider`` mirrors the Python provider order
  (Jenkins → GitHub Actions → CircleCI → Buildkite).
- ``resolve_repository`` branches on the detected provider and
  parses ``BUILDKITE_REPO`` / ``CIRCLE_REPOSITORY_URL`` /
  ``GIT_URL`` via a new ``parse_repo_from_url`` helper that
  recognizes SSH (``git@host:owner/repo[.git]``), HTTP(S), and
  bare ``owner/repo`` shapes — matching the regex pair in
  ``_get_github_repository_from_env``. ``GITHUB_REPOSITORY`` stays
  as the GHA + no-provider fallback.
- ``resolve_pull_request`` reads ``BUILDKITE_PULL_REQUEST`` on
  Buildkite (treating empty / ``false`` as "no PR"); the
  GitHub-event-path branch is unchanged otherwise.

Tests now clear every CI-provider env var the resolver inspects via
a shared ``with_ci_env`` / ``with_ci_env_async`` helper before
applying per-test overrides, so they no longer inherit
provider state from the host (e.g. Buildkite's own CI). Adds 10
new tests covering Buildkite / CircleCI / Jenkins repo resolution,
Buildkite PR resolution edge cases (false / unset / number), and
``parse_repo_from_url`` shape coverage. 23/23 tests green; clippy
clean.

Other recent ``detector.py`` additions (``get_cicd_pipeline_run_url``,
``get_base_ref_name``, ``get_repository_url``, 1-based
``BUILDKITE_RETRY_COUNT``) feed ``junit_processing`` and
``ci git-refs``; neither is ported to Rust on this branch yet, so
nothing to align there.

Fixes INC-1357

Co-Authored-By: Claude Opus 4.7 (1M context) <noreply@anthropic.com>

### CODE REVIEW

**Comment by sileht** [comment]
### Revision history

| # | Type | Changes | Reason | Date |
|---|------|---------|--------|------|
| 1 | initial | `789f791` |  | 2026-05-05 17:25 UTC |
| 2 | content | [`789f791 → 87dba74`](https://github.com/Mergifyio/mergify-cli/compare/c1d15a8290f9dd1113b206c83c3e42b618d2b2cf...87dba742be8c8219acd7cac0520129cb69715d90) ([raw](https://github.com/Mergifyio/mergify-cli/compare/789f79164b3be2f348c05a96d8f28cc972dac78e...87dba742be8c8219acd7cac0520129cb69715d90)) |  | 2026-05-05 17:25 UTC |
| 3 | content | [`87dba74 → a234636`](https://github.com/Mergifyio/mergify-cli/compare/7d1ac9c7ce8ad631e095ae29212053416d542bf1...a2346367276b12a0c8a09c5c3786dbe889aae928) ([raw](https://github.com/Mergifyio/mergify-cli/compare/87dba742be8c8219acd7cac0520129cb69715d90...a2346367276b12a0c8a09c5c3786dbe889aae928)) |  | 2026-05-05 17:28 UTC |
| 4 | content | [`a234636 → 54c69c3`](https://github.com/Mergifyio/mergify-cli/compare/f32f6b85ca09cfcde03abd768c33d1e06de0be9e...54c69c377106c5b9d0b639d178a3e470c5206eb0) ([raw](https://github.com/Mergifyio/mergify-cli/compare/a2346367276b12a0c8a09c5c3786dbe889aae928...54c69c377106c5b9d0b639d178a3e470c5206eb0)) |  | 2026-05-05 17:30 UTC |

File: crates/mergify-ci/src/scopes_send.rs [hunk 1]
```diff
@@ -90,24 +90,118 @@ pub async fn run(opts: ScopesSendOptions<'_>, output: &mut dyn Output) -> Result
...
+
+    if url_str.starts_with("http://") || url_str.starts_with("https://") {
+        let parsed = Url::parse(url_str).ok()?;
+        let path = parsed
+            .path()
+            .trim_start_matches('/')
+            .trim_end_matches('/')
+            .trim_end_matches(".git");
+        return validate_owner_repo(path);
+    }
```
[the remark this answers is not in the document]
Comment (sileht) [reply crates/mergify-ci/src/scopes_send.rs]: Fixed in a234636. The helper has moved to `crates/mergify-ci/src/detector.rs` as `parse_repository_url` and now rejects HTTP(S) URLs whose `query()` or `fragment()` is `Some(_)`, matching Python's end-of-string-anchored regex. Added test cases for `?tab=readme`, `?ref=main` and `#readme` shapes.

**Review by jd — APPROVED** [review]

**Review by JulianMaurin — APPROVED** [review]
